### PR TITLE
Getting the CLI to build on OSX Sierra

### PR DESCRIPTION
**Pull request by livarcocc**
Adding an OverrideRid property used in the GetRuntimeInfo task to override the rid of the CLI. This allows us to build for osx.10.11 when building on osx.10.12.

to build the CLI on sierra, just invoke ./build.sh /p:OverrideRid=osx.10.11-x64

cc @piotrpMSFT @brthor @eerhardt

### CODE REVIEW

File: test/dotnet-publish.Tests/PublishDesktopTests.cs [hunk 1]
```diff
@@ -54,7 +55,9 @@ public PublishDesktopTests()
         [InlineData("KestrelDesktop", "http://localhost:20207", "win7-x86", "libuv.dll", false)]
         public async Task DesktopApp_WithKestrel_WorksWhenPublished(string project, string url, string runtime, string libuvName, bool forceRunnable)
         {
-            var runnable = forceRunnable || string.IsNullOrEmpty(runtime) || (RuntimeEnvironmentRidExtensions.GetLegacyRestoreRuntimeIdentifier().Contains(runtime));
+            var runnable = forceRunnable ||
```
Comment (piotrpMSFT) [inline test/dotnet-publish.Tests/PublishDesktopTests.cs]: *nit/for discusion*
When breaking up a single-line statement joined by binary operators I prefer to make the operator the first element of the new line as this makes the relationship across operands a bit easier to see. 

e.g.
````
   forceRunnable
|| string.IsNullOrEmpty(runtime)
|| ....
&& ....

File: test/dotnet-publish.Tests/PublishTests.cs [hunk 2]
```diff
@@ -58,11 +59,11 @@ public static IEnumerable<object[]> PublishOptions
                 {
                     new object[] { "1", "", "", "", "" },
                     new object[] { "2", "netcoreapp1.0", "", "", "" },
-                    new object[] { "3", "", RuntimeEnvironmentRidExtensions.GetLegacyRestoreRuntimeIdentifier(), "", "" },
+                    new object[] { "3", "", DotnetRuntimeIdentifiers.InferLegacyRestoreRuntimeIdentifier(), "", "" },
```
Comment (piotrpMSFT) [inline test/dotnet-publish.Tests/PublishTests.cs]: RID vs. RuntimeIdentifier? This token is quite verbose...

**Comment by eerhardt** [comment]
:shipit:

File: test/dotnet-publish.Tests/PublishDesktopTests.cs (hunk 1 above)
Comment (livarcocc) [inline test/dotnet-publish.Tests/PublishDesktopTests.cs]: Aha, finally me and Eric finally kind of agree on styling.

**Pull request merged**